### PR TITLE
Add Explanation + C++ Solution for Convention (Silver)

### PR DESCRIPTION
**Pull request by ryanchou-dev**
sorry java people, i couldn't save you...

---
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-858.mdx [hunk 1]
```diff
@@ -134,88 +153,75 @@ public class convention {
...
+#include <cstdio>
+#include <iostream>
+#include <vector>
 
-#include <bits/stdc++.h>
-using namespace std;
-
-#define pb push_back
+using std::endl;
+using std::vector;
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:166]: YOOOO NIIIIICE

File: solutions/usaco-858.mdx [hunk 2]
```diff
@@ -134,88 +153,75 @@ public class convention {
...
 }
 ```
-
 </JavaSection>
-
 <CPPSection>
 
 ```cpp
+#include <algorithm>
+#include <cstdio>
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:161]: is this necessary tho?

File: solutions/usaco-858.mdx [hunk 3]
```diff
@@ -134,88 +153,75 @@ public class convention {
...
-			nBuses++;
+vector<int> arrivals;
+
+bool validate(int k) {
+	// number of buses needed
+	int bus = 0;
+	// current cow
+	int cow = 0;
+	// earliest cow on this bus
+	int lcow = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:177]: (optional) comments on same line as variable declaration

File: solutions/usaco-858.mdx [hunk 4]
```diff
@@ -134,88 +153,75 @@ public class convention {
...
+	arrivals.resize(n);
+	
+	for (int i = 0; i < n; i++) {
+		std::cin >> arrivals[i];
 	}
-	cout << l << endl;
-}
 
+	// sort by arrival time to process cows in order
+	sort(arrivals.begin(), arrivals.end());
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:210]: did you make sure this compiles? you didn't use `namespace std` sooo

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-858.mdx (hunk 4 above)
Comment (ryanchou-dev) [reply solutions/usaco-858.mdx:210]: yep
[stack overflow linky](https://stackoverflow.com/questions/41491995/using-stdsort-without-prefix-std-and-also-without-using-namespace-std-c)

File: solutions/usaco-858.mdx (hunk 2 above)
Comment (ryanchou-dev) [reply solutions/usaco-858.mdx:161]: `algorithm` for sort
`cstdio` for freopen

File: solutions/usaco-858.mdx (hunk 3 above)
Comment (ryanchou-dev) [reply solutions/usaco-858.mdx:177]: sure

File: solutions/usaco-858.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-858.mdx:207]: alr i'll believe you

File: solutions/usaco-858.mdx [hunk 5]
```diff
@@ -134,88 +153,72 @@ public class convention {
...
 
-```
+	int lo = 0, hi = arrivals[n - 1] - arrivals[0];
```
Comment (SansPapyrus683) [inline solutions/usaco-858.mdx:209]: (optional) separate declarations

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/usaco-858.mdx (hunk 5 above)
Comment (ryanchou-dev) [reply solutions/usaco-858.mdx:209]: oops mb

**Review by TruaShamu — APPROVED** [review]

**Review by ar492 — APPROVED** [review]

**Pull request merged**